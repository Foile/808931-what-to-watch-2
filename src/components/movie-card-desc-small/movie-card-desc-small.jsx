import React from "react";
import {string, shape, number, bool} from "prop-types";
import {Link} from "react-router-dom";

const MovieCardDescSmall = (props) => <div className="movie-card__desc">
  <h2 className="movie-card__title">{props.movie.name}</h2>
  <p className="movie-card__meta">
    <span className="movie-card__genre">{props.movie.genre}</span>
    <span className="movie-card__year">{props.movie.released}</span>
  </p>

  <div className="movie-card__buttons">
    <button
      className="btn btn--play movie-card__button"
      type="button"
      onClick = {()=>{
      }}
    >
      <svg viewBox="0 0 19 19" width="19" height="19">
        <use xlinkHref="#play-s"></use>
      </svg>
      <span>Play</span>
    </button>
    <button
      className="btn btn--list movie-card__button"
      type="button"
      onClick = {()=>{
      }}
    >
      <svg viewBox="0 0 19 20" width="19" height="20">
        <use xlinkHref="#add"></use>
      </svg>
      <span>My list</span>
    </button>
    {props.review ? <Link to={`/films/${props.movie.id}/review`} className="btn movie-card__button">Add review</Link> : ``}
  </div>
</div>;

MovieCardDescSmall.propTypes = {review: bool,
  movie: shape({
    id: number.isRequired,
    name: string.isRequired,
    genre: string.isRequired,
    released: number,
    posterImage: string,
    backgroundImage: string
  }).isRequired};


export default MovieCardDescSmall;
