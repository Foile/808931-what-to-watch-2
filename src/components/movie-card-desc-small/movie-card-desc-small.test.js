import React from 'react';
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {MovieCardDescSmall} from './movie-card-desc-small';

it(`MovieCardDescSmall correctly renders after relaunch`, () => {
  const movie = {id: 1, name: `Macbeth`, previewImage: ``, previewVideoLink: ``, genre: `Comedy`};
  const tree = renderer
  .create(<BrowserRouter>
    <MovieCardDescSmall
      movie = {movie}
      review = {true} />
  </BrowserRouter>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
