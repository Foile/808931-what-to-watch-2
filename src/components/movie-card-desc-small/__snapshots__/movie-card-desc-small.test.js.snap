// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardDescSmall correctly renders after relaunch 1`] = `
<div
  className="movie-card__desc"
>
  <h2
    className="movie-card__title"
  >
    Macbeth
  </h2>
  <p
    className="movie-card__meta"
  >
    <span
      className="movie-card__genre"
    >
      Comedy
    </span>
    <span
      className="movie-card__year"
    />
  </p>
  <div
    className="movie-card__buttons"
  >
    <button
      className="btn btn--play movie-card__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="19"
        viewBox="0 0 19 19"
        width="19"
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <button
      className="btn btn--list movie-card__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="20"
        viewBox="0 0 19 20"
        width="19"
      >
        <use
          xlinkHref="#add"
        />
      </svg>
      <span>
        My list
      </span>
    </button>
    <a
      className="btn movie-card__button"
      href="/films/1/review"
      onClick={[Function]}
    >
      Add review
    </a>
  </div>
</div>
`;
